pkg/obfuscator/tracker.go

```diff
@@ -15,11 +15,11 @@ type ReplacementTracker interface {
 	Initialize(replacements map[string]string)
 
 	// Report returns a mapping of strings which were replaced.
-	Report() map[string]string
+	Report() map[string]Replacement
```

Comment:
here's what I would return instead:

```
type ReplacementReport struct {
    replacements []Replacement
}

type Replacement {
   original string
   replaced string
   counter int 
}

// for backwards compatibility in all our tests
func (ReplacementReport) asMap() map[string]string {
}
```

that way we can serialize the list in yaml correctly, can properly sort on the struct level and do diffs in the e2e tests.

Reply:
in the implementation, you can just keep a second map for the counts and on Report() just collate, similar to the defensive copy we do